.github/workflows/gh-ci.yaml

```diff
@@ -108,6 +108,13 @@ jobs:
     - name: run_tests
       if: contains(matrix.name, 'asv_check') != true
       run: |
+        export OPENBLAS_NUM_THREADS=1
+        export GOTO_NUM_THREADS=1
+        export OMP_NUM_THREADS=1
+        export MKL_NUM_THREADS=1
+        # limit to 2 workers to avoid overloading the CI
+        export PYTEST_XDIST_AUTO_NUM_WORKERS=2
```

Comment:
Could this be set to *auto*, because that's what we are already using. Does *auto* not work correctly?

If *auto* does not work then I'd prefer we define a variable GITHUB_CI_MAXCORES or similar and then use it invoking `pytest -n $GITHUB_CI_MAXCORES`. I prefer commandline arguments over env vars when determining code behavior because you immediately see what affects the command itself.

Reply:
I was planning to find a way to find the default number of workers pytest will use and reduce it by one because the Ubuntu runner has 4 cores and Mac has 3 cores. but I ended up setting it to 2 and found the performance acceptable.

Reply:
ok

Reply:
> Ubuntu runner has 4 cores and Mac has 3 cores.

It's unclear from this conversation that altering the value of `-n` is actually resulting in any difference in the issue we're seeing with the the one multiprocessing test failing (i.e. the only cause of timeouts), could you confirm this please?

If it's not affecting things, then my suggestion is to stick with auto unless there's a substantial difference in performance that we've missed.